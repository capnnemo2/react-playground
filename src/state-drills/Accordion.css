.Accordion {
    list-style-type: none;
    margin: 0;
    padding: 0;
    margin-top: 30px;
}

.Accordion_item button {
    background: lightcoral;
    border: 1px solid #222;
    padding: 10px 20px;
    width: 100%;
    font-size: 1em;
    font-weight: bold;
}

.Accordion_item p {
    background: #fdfcfc;
    border: 1px solid #777;
    border-top: none;
    margin: 0;
    padding: 20px 10px;
}