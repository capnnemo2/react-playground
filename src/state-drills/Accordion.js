import React from 'react';



class Accordion extends React.Component {
    render() {
        return (
            <div>
                <ul>
                    <li></li>
                </ul>
            </div>
        )
    }
}

export default Accordion;