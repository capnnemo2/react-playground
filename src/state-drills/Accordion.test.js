import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordion from './Accordion';

describe('Accordion component', () => {
    const sectionsProps = [
        { title: 'Ice Cream',
            content: 'There is an almost infinite variety of ice cream flavors, but there are only really a handful of flavors that matter. King among those select few is Chocolate Chip Cookie Dough.'},
        { title: 'Popsicles',
            content: 'Popsicles seem to be most popular with younger humans. They are a perfect treat for a hot summer day, although they can be a bit messy.'},
        { title: 'Tater Tots',
            content: 'An unconventional dessert choice, tater tots provide steamy potato bits and an opportunity to eat ketchup, barbeque sauce, or the hot sauce of your choosing.'}
    ]

    it('renders without errors', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Accordion />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders an empty li when the sections prop is not supplied', () => {
        const wrapper = shallow(<Accordion />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders no section as active by default', () => {
        const wrapper = shallow(<Accordion sections={sectionsProps} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders a clicked section', () => {
        const wrapper = shallow(<Accordion sections={sectionsProps} />)
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders only the last section when multiple sections have been clicked', () => {
        const wrapper = shallow(<Accordion sections={sectionsProps} />)
        wrapper.find('button').at(1).simulate('click')
        wrapper.find('button').at(2).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })

})