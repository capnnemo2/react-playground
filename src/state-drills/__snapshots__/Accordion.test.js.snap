// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component renders a clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Ice Cream
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Popsicles
    </button>
    <p>
      Popsicles seem to be most popular with younger humans. They are a perfect treat for a hot summer day, although they can be a bit messy.
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Tater Tots
    </button>
  </li>
</ul>
`;

exports[`Accordion component renders an empty li when the sections prop is not supplied 1`] = `<ul />`;

exports[`Accordion component renders no section as active by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Ice Cream
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Popsicles
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Tater Tots
    </button>
  </li>
</ul>
`;

exports[`Accordion component renders only the last section when multiple sections have been clicked 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Ice Cream
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Popsicles
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Tater Tots
    </button>
    <p>
      An unconventional dessert choice, tater tots provide steamy potato bits and an opportunity to eat ketchup, barbeque sauce, or the hot sauce of your choosing.
    </p>
  </li>
</ul>
`;
